src/Sulu/Bundle/WebsiteBundle/Routing/PortalLoader.php

```diff
@@ -22,32 +23,41 @@
  */
 class PortalLoader extends Loader
 {
+    /**
+     * @var WebspaceManagerInterface
+     */
+    private $webspaceManager;
+
+    public function __construct(WebspaceManagerInterface $webspaceManager)
+    {
+        $this->webspaceManager = $webspaceManager;
+    }
+
     public function load($resource, $type = null)
     {
         $collection = new RouteCollection();
 
         /** @var Route[] $importedRoutes */
         $importedRoutes = $this->import($resource, null);
 
-        $condition = \sprintf(
-            'request.get("_sulu").getAttribute("portalInformation") !== null && request.get("_sulu").getAttribute("portalInformation").getType() === %s',
-            RequestAnalyzerInterface::MATCH_TYPE_FULL
-        );
+        $prefixes = [];
+        foreach ($this->webspaceManager->getPortalInformations() as $portalInformation) {
+            // cast null to string as prefix can be empty string
+            $prefixes[] = preg_quote((string) $portalInformation->getPrefix());
+        }
 
         foreach ($importedRoutes as $importedRouteName => $importedRoute) {
-            $importedCondition = $importedRoute->getCondition();
-
             $collection->add(
                 $importedRouteName,
                 new PortalRoute(
                     '{prefix}' . $importedRoute->getPath(),
                     $importedRoute->getDefaults(),
-                    \array_merge(['prefix' => '.*', 'host' => '.+'], $importedRoute->getRequirements()),
+                    \array_merge(['prefix' => implode('|', $prefixes)], $importedRoute->getRequirements()),
```

Comment:
this leads to an error when the result of `implode('|', $prefixes)` is an empty string 😕

Reply:
what error do you get? empty string would be the correct value then.